internal/services/mssql/mssql_server_resource.go

```diff
@@ -266,214 +265,267 @@ func resourceMsSqlServerCreate(d *pluginsdk.ResourceData, meta interface{}) erro
+	future, err := client.CreateOrUpdate(ctx, id, props)
 	if err != nil {
-		return fmt.Errorf("issuing create request for %s: %+v", id.String(), err)
+		return fmt.Errorf("issuing create request for %s: %+v", id, err)
 	}
 
-	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		if response.WasConflict(future.Response()) {
-			return fmt.Errorf("SQL Server names need to be globally unique and %q is already in use", id.Name)
+	if err = future.Poller.PollUntilDone(ctx); err != nil {
+		if response.WasConflict(future.HttpResponse) {
+			return fmt.Errorf("SQL Server names need to be globally unique and %q is already in use", id.ServerName)
 		}
 
-		return fmt.Errorf("waiting for creation/update of %s: %+v", id.String(), err)
+		return fmt.Errorf("waiting for creation of %s: %+v", id, err)
 	}
 
 	d.SetId(id.ID())
 
-	connection := sql.ServerConnectionPolicy{
-		ServerConnectionPolicyProperties: &sql.ServerConnectionPolicyProperties{
-			ConnectionType: sql.ServerConnectionType(d.Get("connection_policy").(string)),
+	connection := serverconnectionpolicies.ServerConnectionPolicy{
+		Properties: &serverconnectionpolicies.ServerConnectionPolicyProperties{
+			ConnectionType: serverconnectionpolicies.ServerConnectionType(d.Get("connection_policy").(string)),
 		},
 	}
-	if _, err = connectionClient.CreateOrUpdate(ctx, id.ResourceGroup, id.Name, connection); err != nil {
-		return fmt.Errorf("issuing create request for Connection Policy %s: %+v", id.String(), err)
+
+	if err = connectionClient.CreateOrUpdateThenPoll(ctx, id, connection); err != nil {
+		return fmt.Errorf("issuing create request for Connection Policy %s: %+v", id, err)
 	}
 
 	return resourceMsSqlServerRead(d, meta)
 }
 
 func resourceMsSqlServerUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).MSSQL.ServersClient
-	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
 	connectionClient := meta.(*clients.Client).MSSQL.ServerConnectionPoliciesClient
 	adminClient := meta.(*clients.Client).MSSQL.ServerAzureADAdministratorsClient
 	aadOnlyAuthenticationsClient := meta.(*clients.Client).MSSQL.ServerAzureADOnlyAuthenticationsClient
 	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id := parse.NewServerID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
+	id, err := commonids.ParseSqlServerID(d.Id())
+	if err != nil {
+		return err
+	}
 
-	existing, err := client.Get(ctx, id.ResourceGroup, id.Name, "")
+	existing, err := client.Get(ctx, *id, servers.DefaultGetOperationOptions())
 	if err != nil {
 		return fmt.Errorf("retrieving %s: %+v", id, err)
 	}
 
-	location := azure.NormalizeLocation(d.Get("location").(string))
-	version := d.Get("version").(string)
-
-	t := d.Get("tags").(map[string]interface{})
-	metadata := tags.Expand(t)
+	if model := existing.Model; model != nil {
```

Comment:
since we're nil-checking this below/relying on this having a value - we can remove this `if` statement:

```suggestion
```

Reply:
Fixed.